packages/react/src/hooks/useDataStore.tsx

```diff
@@ -11,99 +11,107 @@ import {
 } from '../primitives/types/datastore';
 
 /**
- * Perform a collection query against a DataStore model
+ * Perform a query against a DataStore model
+ * @param props object containing params for calling either calling Datastore.query or Datastore.observeQuery
+ * @returns the result of either calling Datastore.query or Datastore.observeQuery for the provided props
  * @internal
  */
-export const useDataStoreCollection = <M extends PersistentModel>({
-  model,
-  criteria,
-  pagination,
-}: DataStoreCollectionProps<M>): DataStoreCollectionResult<M> => {
-  const [result, setResult] = React.useState<DataStoreCollectionResult<M>>({
-    items: [],
-    isLoading: false,
-    error: undefined,
-  });
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props: DataStoreBindingProps<Model, 'record'>
+): DataStoreItemResult<Model>;
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props: DataStoreBindingProps<Model, 'collection'>
+): DataStoreCollectionResult<Model>;
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props:
+    | DataStoreBindingProps<Model, 'record'>
+    | DataStoreBindingProps<Model, 'collection'>
+): DataStoreItemResult<Model> | DataStoreCollectionResult<Model> {
```

Comment:
I'm sad that we did this, but it looks like codegen-ui is dependent on this behavior rather than just the underlying hooks.

Reply:
Actually, I just confirmed we no longer actually use `record`, only `collection` (then we manually unpack items[0] similar to what you're doing in the response for `record`) https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react.test.ts.snap

Reply:
Actually, I just confirmed we no longer actually use `record`, only `collection` (then we manually unpack items[0] similar to what you're doing in the response for `record`) https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react.test.ts.snap

Reply:
Just spitballing here, and probably not pertinent to this PR, but in the future maybe we can consider migrating to only using `useDataStoreCollection`?

Reply:
Agreed, probably something you'll want to chat w/ the Studio UI team about updating on their end, then building a deprecation path here.